### PR TITLE
[PLATFORM-1362] Public stream's inspector fix

### PR DESCRIPTION
**Pull request by mondoreale**
Just like with StreamPage, the preview page is no longer wrapped with `userIsAuthenticated` helper thus doesn't always require login. We rely on permissions now which in general are better indicator of what you can and cannot see or do.

This here is a "quick" fix to a larger issue of public resources being walled off by a "be-a-user" constraint. It'll do for now thought.

I'm open for suggestions, cheers!

### CODE REVIEW

File: app/src/routes/definitions.json [hunk 1]
```diff
@@ -60,10 +60,10 @@
     "terms": "https://s3.amazonaws.com/streamr-public/streamr-terms-of-use.pdf",
     "transactions": "/core/transactions",
     "userpages": {
-        "root": "/core"
+        "root": "/core",
+        "streamPreview": "/core/stream/preview/:streamId"
```
Comment (juhah) [inline app/src/routes/definitions.json:64]: I would prefer `/core/stream/:id/preview`. Our urls are not that consistent, though, but that would be more in line with `/products/:id/streamPreview/:streamId` and  `/core/products/:id/edit`.

File: app/src/shared/hooks/useStreamReadPermission.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    const loaded = !!ready && !fetching
+
+    const canRead = (permissions || []).includes('read')
+
+    if (loaded && !canRead) {
+        throw new UnauthorizedError()
+    }
+
+    return loaded
```
Comment (juhah) [inline app/src/shared/hooks/useStreamReadPermission.js:85]: Would this be more useful as a more generic permission hook, to return all of the stream's permissions?

**Comment by juhah** [comment]
And one more thing: If you access a preview that you don't have access to, we shown an error screen instead of a 404 which might be more approriate (as happens with non-permitted canvases for example):

Screen Shot 2020-05-08 at 10 30 45

File: app/src/routes/definitions.json (hunk 1 above)
Comment (mondoreale) [reply app/src/routes/definitions.json:64]: Yep, `/core/streams/:id/preview` is the way to go. Here I just moved it over from its old location, i.e. 
```json
"userPageStreamPreview": "/core/stream/preview/:streamId",
```
I'll update!

File: app/src/shared/hooks/useStreamReadPermission.js (hunk 2 above)
Comment (mondoreale) [reply app/src/shared/hooks/useStreamReadPermission.js:85]: Good point. I'm getting a strange vibe from this hook… that it isn't really that reusable. 😅

File: app/src/routes/definitions.json (hunk 1 above)
Comment (mondoreale) [reply app/src/routes/definitions.json:64]: I'll leave marketplace's links be for now. We can address that topic later.

File: app/src/marketplace/components/StreamPreviewPage/index.jsx [hunk 3]
```diff
@@ -104,6 +106,16 @@ const StreamPreviewPage = ({
         setHasData(true)
     }, [isMounted])
 
+    const permissions = useStreamPermissions(urlId)
+
+    if (!permissions && !productId) {
```
Comment (mondoreale) [inline app/src/marketplace/components/StreamPreviewPage/index.jsx:111]: If `productId` is defined then we go with the old approach = no permissions check.

File: app/src/userpages/components/StreamLivePreview/index.jsx [hunk 4]
```diff
@@ -48,7 +49,21 @@ const mapDispatchToProps = (dispatch: Function, { match: { params: { streamId }
...
+
+        setTimeout(() => {
+            if (window.location.href !== href) {
+                return
+            }
+
+            dispatch(push(routes.stream({
+                id: streamId,
+            })))
+        }, 100)
```
Comment (mondoreale) [inline app/src/userpages/components/StreamLivePreview/index.jsx:65]: If after 100ms location didn't change then we go to the stream page. That's a one hacky way, yup.

**Pull request merged**